### PR TITLE
Improve map_complete_blocks by adding support for collapsing dims

### PR DESCRIPTION
**Pull request by wjbenfold**
### Description

Not needed for #4676 but this seems a useful feature for map_complete_blocks to have so parking it here as suggested on that PR

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: lib/iris/_lazy_data.py [hunk 1]
```diff
@@ -386,8 +387,20 @@ def map_complete_blocks(src, func, dims, out_sizes):
...
     for dim, size in zip(dims, out_sizes):
-        out_chunks[dim] = size
-
-    return data.map_blocks(func, chunks=out_chunks, dtype=src.dtype)
+        if size == 0:
+            drop_axis.append(dim)
+        else:
+            out_chunks[dim] = size
+
+    out_chunks = [
```
Comment (pp-mo) [inline lib/iris/_lazy_data.py:398]: :+1:
I was sure I could code this more clearly, but everything I tried make it worse !

**Comment by pp-mo** [comment]
@wjbenfold just spotted this again.
What is the driver for it ?
are you ready to take it out of Draft ?

**Comment by wjbenfold** [comment]
> What is the driver for it ?
> are you ready to take it out of Draft ?

As referenced in the description, I implemented it for #4676 but it then turned out to be unnecessary (as chunking in the run dimension was handled fine by dask). It's thus sat here rather than binning something that could well be useful in future, but it's currently got no specific driver so I was planning to leave it in draft. I can do differently if you'd prefer?

**Comment by pp-mo** [comment]
>  I was planning to leave it in draft. I can do differently if you'd prefer?

No, by all means it can wait -- up to you
Just making sure it wasn't obsolete + forgotten

**Pull request closed without merging**